components/MeasureDatePickers.tsx

```diff
@@ -0,0 +1,114 @@
+import { Center, Grid, Text } from "@mantine/core";
+import { DatePicker } from "@mantine/dates";
+import React, { useEffect, useState } from "react";
+import { Calendar, X } from "tabler-icons-react";
+import { cleanNotifications, showNotification } from "@mantine/notifications";
+import { DateTime } from "luxon";
+
+export interface MeasureDatePickerProps {
+  id?: string;
+  periodStart: Date;
+  periodEnd: Date;
+  startOnUpdate: (submittedVal: Date) => void;
+  endOnUpdate: (submittedVal: Date) => void;
+}
+
+/**
+ * MeasureDatePickers is a component that renders two DatePickers inside of a Grid.
+ * They are pre-filled with dates specified as props or, if there is an ID in the router,
+ * then the DatePickers are given the values form that Measure's effectivePeriod
+ * @returns two DatePickers inside a Grid or an error message
+ */
+const MeasureDatePickers = ({
+  id,
+  periodStart,
+  periodEnd,
+  startOnUpdate,
+  endOnUpdate,
+}: MeasureDatePickerProps) => {
+  const [fetchingError, setFetchingError] = useState(false);
+
+  useEffect(() => {
+    if (id) {
+      let fetchStatus = { status: 500, statusText: "Failed fetch request" };
+      //fetch the Measure's JSON body from the test server based on id from url
+      fetch(`${process.env.NEXT_PUBLIC_DEQM_SERVER}/Measure/${id}`)
+        .then((data) => {
+          fetchStatus = { status: data.status, statusText: data.statusText };
+          return data.json();
+        })
+        .then((resourcePageBody) => {
```

Comment:
I think we should be a bit stricter about the TypeScript types here. It will allow us to be safer about accessing effectivePeriod off of the measure, because certain things may be undefined. For now, could we cast this to a `fhirJson.Measure | fhirJson.OperationOutcome` resource since we know that those are the possible responses from the server. Then, we'll have to resolve some of the TypeScript issues that show up, for good reason though!

```suggestion
          return data.json() as Promise<fhirJson.Measure | fhirJson.OperationOutcome>;
        })
        .then((resourcePageBody) => {
```

Some advice:

* Typescript will complain that effectivePeriod does not exist on the type OperationOucome. If you get into the block where response is 200 OK, you _know_ that it must be a measure resources at that point. You should create a new variable like `const measure = resourcePageBody as fhirJson.Measure` and then proceed with the checks
* Similar to above. The else block will complain that `issue` does not exist on type `Measure`. If you get into this else, you know that it must be of type `fhirJson.OperationOutcome`. I'd recommend creating a variable and casting in the else case too like the first bullet.
* The `?` operator in TS is your friend for resolving a lot of these issues